### PR TITLE
How to trade

### PR DESCRIPTION
**Pull request by benlvb**

### CODE REVIEW

File: labs/2-perpetual-exchange/2-trading.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+## Take-Profit / Stop-Loss Orders
+
+You can set take-profit and stop-loss trigger orders by selecting the `Add TP` or `Add SL` button in the position row.
+
+Once you create a trigger order, it will be displayed in your position's row under the `Take Profit` and `Stop Loss` columns. You have the flexibility to edit the order and adjust the trigger price as needed.
+
+In the event of manually closing a position, the associated trigger orders will be automatically canceled and closed, providing a seamless experience when opening new future positions.
+
+*Note: While orders are set, execution is not guaranteed. Several factors, including the mark price (an aggregate of exchange prices), reaching the specified price, and the order's size exceeding the remaining position size, can influence the execution outcome.*
```
Comment (hjchan) [inline labs/2-perpetual-exchange/2-trading.md:67]: TPSL currently will only close the whole position, order's size won't exceed the remaining position size

File: labs/2-perpetual-exchange/2-trading.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+## Liquidations
+
+In the scenario of opening a long position in SOL, where the position size surpasses the collateral value, there exists a critical point known as the Liquidation Price. This price is calculated based on the threshold where the loss amount, collateral value, and borrow fee collectively dip below 1% of your position's size. If the token's price surpasses this point, the system will automatically close the position.
```
Comment (hjchan) [inline labs/2-perpetual-exchange/2-trading.md:71]: 0.5% of your position's size

File: labs/2-perpetual-exchange/2-trading.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+It's crucial to note that the liquidation price is subject to change over time, particularly with leverage exceeding 10x and extended position durations. Regularly monitoring your liquidation price is essential. To mitigate the risk of liquidation, collateral adjustments and leverage fine-tuning can be performed through the `Edit` button in the position row, offering an avenue to add collateral and enhance the liquidation price.
+
+Upon liquidation, any remaining collateral, net of losses and fees, will be returned to your account.
+
+## Pricing
+
+On Jupiter Perpetual Exchange, **trades incur no price impact**, allowing for the execution of large trades precisely at the mark price. However, in periods of elevated volatility, a spread may occur between the Pyth pricing and the median price of reference exchanges.
```
Comment (hjchan) [inline labs/2-perpetual-exchange/2-trading.md:79]: no spread and median price

File: labs/2-perpetual-exchange/2-trading.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+Mark prices are conveniently presented in the same row as the market name *(e.g., SOL-PERP)*. Notably, long positions are initiated at the higher price and closed at the lower price, while short positions are opened at the lower price and closed at the higher price.
```
Comment (hjchan) [inline labs/2-perpetual-exchange/2-trading.md:81]: long and short position are opened and closed at the same price

File: labs/2-perpetual-exchange/2-trading.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+Jupiter Perpetuals Trading seamlessly integrates Jupiter Swap. Consequently, when a swap is required during the opening or closing of a position, regular DEX fees and slippages are applicable. The specific fee is contingent on whether the swap enhances or diminishes the balance.
+
+### Execution Fee
+
+The process of opening, closing, or editing a position on Jupiter Perpetual Exchange entails two transactions:
+
+- Users initiate the first transaction to request actions such as opening, closing, depositing collateral, or withdrawing collateral.
+
+- This transaction is directed to an escrow account ([PDA](https://solanacookbook.com/core-concepts/pdas.html#facts)), which monitors the blockchain for these requests and executes them as needed.
```
Comment (hjchan) [inline labs/2-perpetual-exchange/2-trading.md:97]: keeper monitors the requests and executes them

**Comment by benlvb** [comment]
@hjchan @melvinzhang can take a quick look and help review?

File: labs/2-perpetual-exchange/2-trading.md [hunk 6]
```diff
@@ -92,6 +92,6 @@ Jupiter Perpetuals Trading seamlessly integrates Jupiter Swap. Consequently, whe
 
 The process of opening, closing, or editing a position on Jupiter Perpetual Exchange entails two transactions:
 
-- Users initiate the first transaction to request actions such as opening, closing, depositing collateral, or withdrawing collateral.
+- Users initiate the first transaction to the keeper to request actions such as opening, closing, depositing collateral, or withdrawing collateral.
```
Comment (hjchan) [inline labs/2-perpetual-exchange/2-trading.md:95]: user didn't initiate transaction to the keeper, it send onchain and the keeper monitor it

File: labs/2-perpetual-exchange/2-trading.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+Mark prices are conveniently presented in the same row as the market name *(e.g., SOL-PERP)*. Notably, long positions and short positions are opened and closed at the same price as `Mark Price` when initiated market order. 
+
+## Fees
+
+Opening or closing a position on Jupiter Perpetual Exchange incurs a fee equivalent to 0.1% of the position size.
+
+For long positions, the collateral is the token being longed *(e.g., SOL for SOL longs, BTC for BTC longs)*, while short positions are collateralized with supported stablecoins like USDC or USDT.
+
+Jupiter Perpetuals Trading seamlessly integrates Jupiter Swap. Consequently, when a swap is required during the opening or closing of a position, regular DEX fees and slippages are applicable. The specific fee is contingent on whether the swap enhances or diminishes the balance.
```
Comment (melvinzhang) [inline labs/2-perpetual-exchange/2-trading.md:89]: > contingent on whether the swap enhances or diminishes the balance

don't quite understand this part

File: labs/2-perpetual-exchange/3-jlp-pool.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
+---
+sidebar_label: "JLP Pool"
+title: "JLP Pool"
+description: How to Become Liquidity Provider
+---
+
+Users have the opportunity to become Liquidity Providers (LPs) by allocating their assets or tokens into the Jupiter Liquidity Provider Pool (JLP Pool) and in return, receive 70% of the fees, compounding into the price of JLP.
```
Comment (melvinzhang) [inline labs/2-perpetual-exchange/3-jlp-pool.md:7]: have the opportunity to -> can

File: labs/2-perpetual-exchange/3-jlp-pool.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+### JLP, Liquidity Provider Token
+
+JLP, or the Liquidity Provider token, represents a meticulously constructed index of assets designed for swaps and leverage trading. These tokens can be created using any index asset and later redeemed for any index asset.
+
+Liquidity providers serve a crucial role as they act as counterparts to traders. When traders seek to open leverage positions, they borrow tokens from the pool. Liquidity providers, in return, earn fees from these leverage trading activities, along with borrowing fees and earnings from swaps. As a JLP holder, you receive 70% of the fees generated by the trading exchange. The exchange generates fees and yield in three ways:
```
Comment (melvinzhang) [inline labs/2-perpetual-exchange/3-jlp-pool.md:13]: counterparts -> counterparty

File: labs/2-perpetual-exchange/3-jlp-pool.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+2. JLP Pool will automatically identify the token asset in the JLP pool with the highest weightage and swap your chosen asset into that pool if your selected output token is not in the list of token assets. Your assets will then be withdrawn directly into your wallet.
+3. It's important to note that when you withdraw from the JLP token pool with a relatively low weightage, fees will be incurred. Conversely, withdrawing from a token pool with higher weightage will result in lower or even zero fees during the withdrawal process.
+4. In situations where users are withdrawing a token asset not present in the pool (e.g., SOL, ETH, BTC, USDC, or USDT), JLP will withdraw from the token pool with the highest weightage, incurring minimal fees. The withdrawn assets will be swapped into the selected output token and then transferred directly into the user's wallet.
+
+![JLP2](../img/jlp2.jpg)
+
+### Target Ratio and Fees
+
+In the JLP pool, every token holds a specific target ratio or weightage.
```
Comment (melvinzhang) [inline labs/2-perpetual-exchange/3-jlp-pool.md:62]: holds -> has

File: labs/2-perpetual-exchange/3-jlp-pool.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+
+The transactions involving the addition or removal of liquidity have the primary purpose of adjusting a token's ratio in the pool to align it more closely with the predefined target.
+
+Similarly, each custody plays a role in maintaining a target ratio within the pool. This target ratio significantly influences the fees that are applicable to instructions aimed at modifying the ratio of the custodies within the pool.
```
Comment (melvinzhang) [inline labs/2-perpetual-exchange/3-jlp-pool.md:66]: > Similarly, each custody plays a role in maintaining a target ratio within the pool.

This sentence can be removed without affecting the info as it is implementation detail.

> This target ratio significantly ...

This is the key part but this sentence is a bit convoluted. One possible alternative. "Instructions that move the token's ratio away from the target incurs additional fees while instructions that move it closer to the target get a fee rebate."

Comment (benlvb) [reply labs/2-perpetual-exchange/3-jlp-pool.md:66]: Should we use the word `fee rebate`?

Comment (benlvb) [reply labs/2-perpetual-exchange/3-jlp-pool.md:66]: Or fee discount?

Comment (melvinzhang) [reply labs/2-perpetual-exchange/3-jlp-pool.md:66]: either should be fine, discount seems easier to understand

**Review by melvinzhang — APPROVED** [review]

File: labs/2-perpetual-exchange/2-trading.md (hunk 7 above)
Comment (benlvb) [reply labs/2-perpetual-exchange/2-trading.md:89]: This part is a bit confusing yes. Let me remove it.

**Pull request merged**